app/services/web_mentions/web_mention_handler.rb

```diff
@@ -0,0 +1,41 @@
+module WebMentions
+  class WebMentionHandler
+    def initialize(canonical_url:, article_url: nil, webmention_url: nil)
+      @canonical_url = canonical_url
+      @article_url = article_url
+      @webmention_url = webmention_url
+    end
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def call
+      return if webmention_url.blank?
+
+      send_webmention
+    end
+
+    def webmention_url
+      document = Nokogiri::HTML.parse(HTTParty.get(@canonical_url)).at_css("link[rel='webmention']")
+
+      if document
+        @webmention_url = if Addressable::URI.parse(document["href"]).relative?
+                            Addressable::URI.parse(@canonical_url).origin + document["href"]
+                          else
+                            document["href"]
+                          end
+      end
+    rescue StandardError => e
+      raise e, "Couldn't find the webmention URL of: #{@canonical_url}"
+    end
+
+    private
+
+    def send_webmention
+      HTTParty.post(@webmention_url, "source": @article_url, "target": @canonical_url)
+    rescue HTTParty::Error => e
+      raise e, "Couldn't send webmentions to: #{@webmention_url}"
```

Comment:
I am not very familiar with sending webmentions. What kinds of errors might we see here? Are they likely going to be errors that are worth retrying or not?

Reply:
The main error we could see is the endpoint getting removed or temporarily being unavailable.

Reply:
> the endpoint getting removed

We should probably handle a 404(I'm assuming that's the error we would get) more intelligently so we are not constantly retrying if an endpoint is removed.